Core.ApplicationServices/Contract/Write/ItContractWriteService.cs

```diff
@@ -158,10 +162,52 @@ public class ItContractWriteService : IItContractWriteService
                 .Bind(updateContract => updateContract.WithOptionalUpdate(parameters.PaymentModel, UpdatePaymentModelParameters))
                 .Bind(updateContract => updateContract.WithOptionalUpdate(parameters.ExternalReferences, UpdateExternalReferences))
                 .Bind(updateContract => updateContract.WithOptionalUpdate(parameters.Roles, UpdateRoles))
-                .Bind(updateContract => updateContract.WithOptionalUpdate(parameters.Termination, UpdateTermination))
-                .Bind(updateContract => updateContract.WithOptionalUpdate(parameters.AgreementPeriod, UpdateAgreementPeriod));
+                .Bind(updateContract => updateContract.WithOptionalUpdate(parameters.AgreementPeriod, UpdateAgreementPeriod))
+                .Bind(updateContract => updateContract.WithOptionalUpdate(parameters.Payments, UpdatePayments))
+                .Bind(updateContract => updateContract.WithOptionalUpdate(parameters.Termination, UpdateTermination));
         }
 
+        private Result<ItContract, OperationError> UpdatePayments(ItContract contract, ItContractPaymentDataModificationParameters parameters)
+        {
+            return contract
+                .WithOptionalUpdate(parameters.ExternalPayments, UpdateExternalPayments)
+                .Bind(updatedContract => updatedContract.WithOptionalUpdate(parameters.InternalPayments, UpdateInternalPayments));
+        }
+
+        private Maybe<OperationError> UpdateInternalPayments(ItContract contract, IEnumerable<ItContractPayment> newPaymentState)
+        {
+            return ReplacePayments(true, contract.InternEconomyStreams.ToList(), newPaymentState, contract.ResetInternalEconomyStreams, contract.AddInternalEconomyStream);
+        }
+
+        private Maybe<OperationError> UpdateExternalPayments(ItContract contract, IEnumerable<ItContractPayment> newPaymentState)
+        {
+            return ReplacePayments(false, contract.ExternEconomyStreams.ToList(), newPaymentState, contract.ResetExternalEconomyStreams, contract.AddExternalEconomyStream);
+        }
+
+        private delegate Maybe<OperationError> AddPaymentFunc(Guid? optionalOrganizationUnitUuid, int acquisition, int operation, int other, string accountingEntry, TrafficLight auditStatus, DateTime? auditDate, string note);
+        private Maybe<OperationError> ReplacePayments(
```

Comment:
Måske lige et linjeskift her mellem 187 og 188. Så er det lidt lettere at læse det er to forskellige ting.